app/models/company.rb

```diff
@@ -1,14 +1,71 @@
+# == Schema Information
+#
+# Table name: companies
+#
+#  id                :integer          not null, primary key
+#  name              :string
+#  domain            :string
+#  youtube           :string
+#  twitter           :string
+#  linkedincompany   :string
+#  facebook          :string
+#  angellist         :string
+#  owler             :string
+#  crunchbasecompany :string
+#  pinterest         :string
+#  google            :string
+#  klout             :string
+#  overview          :string
+#  founded           :integer
+#  approx_employees  :integer
+#  created_at        :datetime         not null
+#  updated_at        :datetime         not null
+#  user_id           :integer
+#  logo              :string
+#
+
 class Company < ApplicationRecord
   VALID_DOMAIN_REGEX = /\A[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,6}\z/ix
 
   scope :ordered, -> { order(name: :asc) }
 
-  belongs_to :user
-  belongs_to :tenant, optional: true
-  has_many :pages, dependent: :destroy
+  belongs_to              :user
+  belongs_to              :tenant, optional: true
+  has_many                :pages, dependent: :destroy
+  has_many                :videos, as: :videoable, dependent: :destroy
   has_and_belongs_to_many :industries, -> { distinct }
+
+  mount_uploader :logo, CompanyLogoUploader
+
+  after_create :create_video
+
   validates :name, presence: true, length: { minimum: 3, maximum: 64 }
   validates :domain, presence: true, length: { minimum: 3, maximum: 64 },
                      format: { with: VALID_DOMAIN_REGEX }
-  mount_uploader :logo, CompanyLogoUploader
+
+  private
+
+  def create_video
+    if youtube.present?
```

Comment:
all code related to video must be moved to separate class, e.g. "CompanyVideoProcessor", etc.